doc(faq): Add entry about how rsync handles open files

johanR04 opened the pull request:
Fix #1381

johanr001 committed:
Added FaQ entry for rsync open files (Issue #1381)

johanR04 committed:
Added FaQ entry for rsync open files (Issue #1381)

buhtz commented on gcm-diagnose.log:
```diff
@@ -0,0 +1,197 @@
+﻿Diagnose log at 2025-07-10T15:20:34Z
```
What is this file about?

buhtz commented on FAQ.md:
```diff
@@ -530,6 +530,28 @@ If you encounter clear rules about configuring Samba that it works with
+
+**Explanation**
+
+Back in Time uses rsync to copy the files and folders specified to be backed up in the configuration. Rsync does not lock any files that are open or being modified and therefore the backup can be copied in an inconsistent state. Rsync only reads a file on time when it goes through it and as a result of this only some changes are captured by rsync. This can affect files such as logs, browser cache ...
```
I suggest to wrap the lines at 79 chars. This makes it easier to diff and edit them.

```suggestion
Back In Time uses rsync to copy the files and directories specified to be
backed up in the configuration. Rsync does not lock any files that are open
or being modified and therefore the backup can be copied in an inconsistent
state. Rsync only reads a file on time when it goes through it and as a result
of this only some changes are captured by rsync. This can affect files such as
logs, browser caches, databases or virtual machine images where inconsistencies
can even lead to data corruption.
```

buhtz commented on FAQ.md:
```diff
@@ -530,6 +530,28 @@ If you encounter clear rules about configuring Samba that it works with
+**To reduce this risk, the following approaches can be considered:**
+
+- **Filesystem snapshots**
+   If using a filesystem like btrfs and ZFS that has a snapshot function this can be used together with Back in Time. Filesystem snapshots provide a read-only copy of a filesystem frozen at a specific point in time, which ensures data integrity even for open/changing files. Configure Back in Time to backup from this filesystem's read-only snapshot.
```
```suggestion
   If using a filesystem like btrfs and ZFS that has a snapshot function this
   can be used together with Back in Time. Filesystem snapshots provide a
   read-only copy of a filesystem frozen at a specific point in time, which
   ensures data integrity even for open/changing files. Configure Back In Time
   to backup from this filesystem's read-only snapshot.
```

buhtz commented on FAQ.md:
```diff
@@ -530,6 +530,28 @@ If you encounter clear rules about configuring Samba that it works with
+   If using a filesystem like btrfs and ZFS that has a snapshot function this can be used together with Back in Time. Filesystem snapshots provide a read-only copy of a filesystem frozen at a specific point in time, which ensures data integrity even for open/changing files. Configure Back in Time to backup from this filesystem's read-only snapshot.
+
+- **Use exclusions**
+   If the filesystem does not have filesystem snapshots available, one solution could be to exclude files that are frequently open or actively changing. The command lsof in Linux presents open files and the processes that opened them as a list. Use this list as base for configuring BIT exclusion list.
```
```suggestion
   If the filesystem does not have filesystem snapshots available, one
   solution could be to exclude files that are frequently open or actively
   changing. The command `lsof` in GNU/Linux presents open files and the
   processes that opened them as a list. Use this list as base for
   configuring BIT exclusion list.
```

buhtz commented on FAQ.md:
```diff
@@ -530,6 +530,28 @@ If you encounter clear rules about configuring Samba that it works with
+   If the filesystem does not have filesystem snapshots available, one solution could be to exclude files that are frequently open or actively changing. The command lsof in Linux presents open files and the processes that opened them as a list. Use this list as base for configuring BIT exclusion list.
+
+- **Application specific handling**
+   For applications that opens and modifies files frequently like databases or virtual machines, specific solutions my be needed. Use the databases own backup function to create a consistent copy and include that in the BIT backup. Virtual machines products typically have ability to create snapshots of their state, that can be included in BIT.
```
```suggestion
   For applications that opens and modifies files frequently like databases
   or virtual machines, specific solutions may be needed. Use the databases
   own backup function to create a consistent copy and include that in the
   BIT backup. Virtual machines products typically have ability to create
   snapshots of their state, that can be included in BIT.
```

buhtz reviewed (changes requested):
Hello,
On behalf of the team, thank you for your contribution and taking the time to improve Back In Time. We appreciate it.

Your PR looks fine. I just made some minor suggestions.
Press "Commit suggestion" if you agree with them.

Additionally I suggest to add a TOC entry for your entry.

Best,
Christian

johanR04 committed:
Update FAQ.md

Co-authored-by: Christian Buhtz <c.buhtz@posteo.jp>

johanR04 committed:
Update FAQ.md

Co-authored-by: Christian Buhtz <c.buhtz@posteo.jp>

johanR04 committed:
Update FAQ.md

Co-authored-by: Christian Buhtz <c.buhtz@posteo.jp>

johanR04 committed:
Update FAQ.md

Co-authored-by: Christian Buhtz <c.buhtz@posteo.jp>

buhtz commented:
btw: Please add the file name `gcm-diagnose.log` to the `.gitignore` file in the root of the repo.

johanR04 committed:
Added TOC entry for FaQ on handling open files

johanR04 committed:
Added TOC entry for FaQ on handling open files

johanR04 commented:
Thank you, I have now corrected according to your recommendations including the TOC entry. I also removed the unnecessary gcm-diagnose.log file. It is a file that was accidentally created when I ran a diagnose command of my git credential manager that was not working correctly.

buhtz reviewed (approved):
Hello Johan,
On behalf of the team, thank you for your contribution and taking
the time to improve Back In Time. We appreciate it.

Your PR looks fine. It will get merged after a cool off period.
To find out how we usually tread PRs please read
["What happens after you opened a PullRequest (PR)?"](https://github.com/bit-team/backintime/blob/dev/CONTRIBUTING.md#what-happens-after-you-opened-a-pull-request-PR).

Best,
Christian
